tests/unit/test_atoi_strtou_noneg.c

```diff
@@ -0,0 +1,125 @@
+// SPDX-FileCopyrightText: 2023, Alejandro Colomar <alx@kernel.org>
+// SPDX-License-Identifier: BSD-3-Clause
+
+
+#include <errno.h>
+#include <limits.h>
+#include <stddef.h>
+#include <stdlib.h>
+
+#include <stdarg.h>  // Required by <cmocka.h>
+#include <stddef.h>  // Required by <cmocka.h>
+#include <setjmp.h>  // Required by <cmocka.h>
+#include <stdint.h>  // Required by <cmocka.h>
+#include <cmocka.h>
+
+#include "atoi/strtou_noneg.h"
+
+
+static void test_strtoul(void **state);
+static void test_strtoul_noneg(void **state);
+static void test_strtoull(void **state);
+static void test_strtoull_noneg(void **state);
+
+
+int
+main(void)
+{
+    const struct CMUnitTest  tests[] = {
+        cmocka_unit_test(test_strtoul),
+        cmocka_unit_test(test_strtoul_noneg),
+        cmocka_unit_test(test_strtoull),
+        cmocka_unit_test(test_strtoull_noneg),
+    };
+
+    return cmocka_run_group_tests(tests, NULL, NULL);
+}
+
+
+static void
+test_strtoul(void **state)
```

Comment:
This API isn't part of our code, so why are you testing it?

Reply:
To show the difference between strtoul(3) and strtoul_noneg().

Reply:
Anyway, I have patches for removing this API and also the tests.  <https://github.com/shadow-maint/shadow/pull/895>

<https://github.com/shadow-maint/shadow/pull/895/commits/b54d5fefe94aee357bff7246e01ef68d03e38798>
<https://github.com/shadow-maint/shadow/pull/895/commits/bce02f881e349d52858a1dc7aa1d580e061fcdec>

Reply:
That code belongs to another project so I'd prefer to avoid testing it. The test might fail in the future and there would be nothing to do on our side. If you still want to keep, at least add a comment stating that this is an external API.

Reply:
Ok, I'll drop those tests.